server/server/data_loader/arangoload.py

```diff
@@ -183,6 +194,100 @@ def process_division_files(
     docs.append({'uid': 'orphan', '_key': 'orphan'})
 
 
+def process_names_files(
+        names_files: List[Path],
+        root_languages: Dict[str, str],
+        super_extra_info:  Dict[str, Dict[str, str]],
+        text_extra_info:  Dict[str, Dict[str, str]]
+) -> List[dict]:
+    """
+    Method for processing name files from sc-data/structure/name
+
+    Args:
+        names_files - list of name Path objects to files from name folder
+        root_languages - parsed data from language.json
+        super_extra_info - parsed data from super_extra_info.json
+        text_extra_info - parsed data from text_extra_info.json
+
+    Returns:
+        list of processed data
+    """
+    docs = []
+    names_files.sort(key=lambda path: len(path.parts))
+    for name_file in names_files:
+        entries: Dict[str, str] = json_load(name_file)
+        for uid, name in entries.items():
+            if not name or type(name) != str:
+                continue
+            extra_info = super_extra_info if uid in super_extra_info else text_extra_info
+            entry = {
+                'uid': uid,
+                '_key': uid,
+                'name': name,
+                'root_lang': root_languages.get(uid, None),
+                'volpage': extra_info.get(uid, {}).get('volpage', None),
+                'biblio_uid': extra_info.get(uid, {}).get('biblio_uid', None),
+                'acronym': extra_info.get(uid, {}).get('acronym', None)
+            }
+
+            docs.append(entry)
```

Comment:
Seems quite complex, also I can see it take the responsibility of filter function and you've done here some kind of mapping. I'm not the python guy but maybe you can use [map](https://docs.python.org/3/library/functions.html#map) and [filter](https://docs.python.org/3/library/functions.html#filter) instead of raw loop.

Reply:
In python you'd totally use for loops for that

Reply:
Is there any reason to not use map and filter function?